### PR TITLE
compare sbom and collected_sources manifest

### PR DESCRIPTION
**Pull request by yash-zededa**
- This will install the compare-sbom-sources utility
- Will compare the output of the sbom and the collected_sources target

### CODE REVIEW

**Review by deitch — CHANGES_REQUESTED** [review]
Can you also add it to the `assets.yml` pipeline, so the results are shown there when run?

**Review by deitch — APPROVED** [review]

**Comment by rene** [comment]
@yash-zededa , it looks like you have squashed some commits during development that have concatenated a lot of "minor fixes" in your commit message. Please, correct them. Let just a summary of the changes you are doing and just one Signed-off in your final commit.

**Comment by yash-zededa** [comment]
@rene thanks for pointing out. This is fixed.

**Review by eriknordmark — COMMENTED** [review]
@deitch I see an 		os.Exit(1)
in compare-sbom-sources, but I can't tell wheher that happens when the comparison fails or only for more serious errors.

Also, the command function has log.Fatal which seems odd; shouldn't it return the error to the caller and have the caller fatal/exit?

File: Makefile [hunk 1]
```diff
@@ -651,6 +656,20 @@ $(COLLECTED_SOURCES): $(ROOTFS_TAR) $(GOSOURCES)| $(INSTALLER)
...
 
+$(COMPARESOURCES):
+	$(QUIET): $@: Begin
+	$(shell GOBIN=$(BUILDTOOLS_BIN) GO111MODULE=on CGO_ENABLED=0 go install $(COMPARE_SOURCE)@$(COMPARESOURCES_VERSION))
+	@echo Done building packages
+	$(QUIET): $@: Succeeded
+
+compare_sbom_collected_sources: $(COLLECTED_SOURCES) $(SBOM) | $(COMPARESOURCES)
+	$(QUIET): $@: Begin
+	$(COMPARESOURCES) $(COLLECTED_SOURCES):collected_sources_manifest.csv $(SBOM)
```
[the remark this answers is not in the document]
Comment (deitch) [reply Makefile:667]: Unless the compare fails fully, then, yes, it would exit 0. If the compare has an error, it exits with 1.

**Comment by deitch** [comment]
@eriknordmark the `os.Exit(1)` is correct; if it cannot execute the command, exit 1.

The `log.Fatal`, on the other hand, shouldn't be there. I will open a PR for it.

**Comment by deitch** [comment]
See #3195

**Review by eriknordmark — APPROVED** [review]
LGTM

**Pull request merged**